src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpInstantiateArgumentExceptionsCorrectly.Fixer.cs

```diff
@@ -13,5 +20,77 @@ namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
     [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
     public sealed class CSharpInstantiateArgumentExceptionsCorrectlyFixer : InstantiateArgumentExceptionsCorrectlyFixer
     {
+        protected override void PopulateCodeFix(CodeFixContext context, Diagnostic diagnostic, string paramPositionString, SyntaxNode node)
+        {
+            if (node is ObjectCreationExpressionSyntax creation)
```

Comment:
It seems like lot of this code can be moved to the base type `InstantiateArgumentExceptionsCorrectlyFixer` if you operate using `IOperation` APIs instead of syntax. You could likely also delete the C# and VB specific implementations, seal `InstantiateArgumentExceptionsCorrectlyFixer` and move `[ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared] to that type.

Reply:
Having scanned the file, I think you should be able to make the implementation completely language-agnostic by switching to IOperation. Just invoke `semanticModel.GetOperation(node, cancellationToken)` to go from syntax node to operation, and then change this check to `if (node is IObjectCreationOperation creation)`

Reply:
Thanks, I will try that

Reply:
It did work after converting the SyntaxNode to IOperation using  SemanticModel

Reply:
Yay!!